src/Microsoft.IdentityModel.Protocols.SignedHttpRequest/SignedHttpRequestHandler.cs

```diff
@@ -571,10 +585,29 @@ internal virtual async Task<SecurityKey> ValidateSignatureAsync(JsonWebToken sig
             if (popKey == null)
                 throw LogHelper.LogArgumentNullException(nameof(popKey));
 
+            if (popKey.CryptoProviderFactory == null)
```

Comment:
CryptoProviderFactory on the key can't be null.
https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/d2512b8f5fc636eaf199e198af6aaa5fa6a8a098/src/Microsoft.IdentityModel.Tokens/SecurityKey.cs#L84

https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/d2512b8f5fc636eaf199e198af6aaa5fa6a8a098/src/Microsoft.IdentityModel.Tokens/SecurityKey.cs#L42-L56

Reply:
correct, just a defensive move.